File: Archives/VB.NET/Simple v8.1/Form1.vb
'/////////////////////////////////////////////////////////////////////////////
'// Form1.vb - VB.NET simple image capture example
'//
'// SnagIt COM server example code
'//
'// Support e-mail: support@techsmith.zendesk.com
'//
'// Copyright 2003-2014 TechSmith Corporation.  All rights reserved.
'/////////////////////////////////////////////////////////////////////////////

Public Class Form1
   Inherits System.Windows.Forms.Form

   ' The reference to the SnagIt COM object was created by
   ' right-clicking on "References" in the solution explorer, and choosing
   ' "Add Reference". Then, under the COM tab, find "SnagIt 1.0 Type Library".
   ' Press "Select" here and click OK. Now VB.NET knows about the SnagIt COM
   ' object and recognizes the functions/properties held in SNAGITLib

   ' Here we declare a global ImageCapture object. This instantiates an 
   ' instance of the COM server. This cannot be local to a function, or it 
   ' would be destroyed before the capture was completed!
   '
   ' Note we are using IImageCapture2 to get new functionality in SnagIt 8.1
   Dim SnagImg As SNAGITLib.IImageCapture2 = New SNAGITLib.ImageCaptureClass()

#Region " Windows Form Designer generated code "

   Public Sub New()
      MyBase.New()

      'This call is required by the Windows Form Designer.
      InitializeComponent()

      'Add any initialization after the InitializeComponent() call

   End Sub

   'Form overrides dispose to clean up the component list.
   Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
      If disposing Then
         If Not (components Is Nothing) Then
            components.Dispose()
         End If
      End If
      MyBase.Dispose(disposing)
   End Sub

   'Required by the Windows Form Designer
   Private components As System.ComponentModel.IContainer

   'NOTE: The following procedure is required by the Windows Form Designer
   'It can be modified using the Windows Form Designer.  
   'Do not modify it using the code editor.
   Friend WithEvents BtnWindow As System.Windows.Forms.Button
   Friend WithEvents GroupBox1 As System.Windows.Forms.GroupBox
   Friend WithEvents ChkPreviewWindow As System.Windows.Forms.CheckBox
   Friend WithEvents GroupBox2 As System.Windows.Forms.GroupBox
   Friend WithEvents BtnRegion As System.Windows.Forms.Button
   Friend WithEvents Label1 As System.Windows.Forms.Label
   Friend WithEvents ChkIncludeCursor As System.Windows.Forms.CheckBox
   Friend WithEvents ChkKeepLinks As System.Windows.Forms.CheckBox
   <System.Diagnostics.DebuggerStepThrough()> Private Sub InitializeComponent()
      Me.BtnWindow = New System.Windows.Forms.Button()
      Me.ChkIncludeCursor = New System.Windows.Forms.CheckBox()
      Me.ChkPreviewWindow = New System.Windows.Forms.CheckBox()
      Me.GroupBox1 = New System.Windows.Forms.GroupBox()
      Me.GroupBox2 = New System.Windows.Forms.GroupBox()
      Me.BtnRegion = New System.Windows.Forms.Button()
      Me.Label1 = New System.Windows.Forms.Label()
      Me.ChkKeepLinks = New System.Windows.Forms.CheckBox()
      Me.GroupBox1.SuspendLayout()
      Me.GroupBox2.SuspendLayout()
      Me.SuspendLayout()
      '
      'BtnWindow
      '
      Me.BtnWindow.Location = New System.Drawing.Point(16, 16)
      Me.BtnWindow.Name = "BtnWindow"
      Me.BtnWindow.Size = New System.Drawing.Size(96, 24)
      Me.BtnWindow.TabIndex = 0
      Me.BtnWindow.Text = "Window Capture"
      '
      'ChkIncludeCursor
      '
      Me.ChkIncludeCursor.Location = New System.Drawing.Point(16, 24)
      Me.ChkIncludeCursor.Name = "ChkIncludeCursor"
      Me.ChkIncludeCursor.TabIndex = 1
      Me.ChkIncludeCursor.Text = "Include Cursor"
      '
      'ChkPreviewWindow
      '
      Me.ChkPreviewWindow.Location = New System.Drawing.Point(16, 48)
      Me.ChkPreviewWindow.Name = "ChkPreviewWindow"
      Me.ChkPreviewWindow.Size = New System.Drawing.Size(112, 24)
      Me.ChkPreviewWindow.TabIndex = 2
      Me.ChkPreviewWindow.Text = "Preview Window"
      '
      'GroupBox1
      '
      Me.GroupBox1.Controls.AddRange(New System.Windows.Forms.Control() {Me.ChkPreviewWindow, Me.ChkIncludeCursor})
      Me.GroupBox1.Location = New System.Drawing.Point(144, 16)
      Me.GroupBox1.Name = "GroupBox1"
      Me.GroupBox1.Size = New System.Drawing.Size(136, 104)
      Me.GroupBox1.TabIndex = 3
      Me.GroupBox1.TabStop = False
      Me.GroupBox1.Text = "Options"
      '
      'GroupBox2
      '
      Me.GroupBox2.Controls.AddRange(New System.Windows.Forms.Control() {Me.BtnRegion, Me.BtnWindow})
      Me.GroupBox2.Location = New System.Drawing.Point(8, 16)
      Me.GroupBox2.Name = "GroupBox2"
      Me.GroupBox2.Size = New System.Drawing.Size(128, 80)
      Me.GroupBox2.TabIndex = 4
      Me.GroupBox2.TabStop = False
      Me.GroupBox2.Text = "Capture Type"
      '
      'BtnRegion
      '
      Me.BtnRegion.Location = New System.Drawing.Point(16, 48)
      Me.BtnRegion.Name = "BtnRegion"
      Me.BtnRegion.Size = New System.Drawing.Size(96, 23)
      Me.BtnRegion.TabIndex = 1
      Me.BtnRegion.Text = "Region Capture"
      '
      'Label1
      '
      Me.Label1.Location = New System.Drawing.Point(16, 128)
      Me.Label1.Name = "Label1"
      Me.Label1.Size = New System.Drawing.Size(264, 40)
      Me.Label1.TabIndex = 5
      Me.Label1.Text = "For simplicity, we will always output to a file, the name of which will be suppli" & _
      "ed by the user when prompted."
      '
      'ChkKeepLinks
      '
      Me.ChkKeepLinks.Location = New System.Drawing.Point(160, 88)
      Me.ChkKeepLinks.Name = "ChkKeepLinks"
      Me.ChkKeepLinks.Size = New System.Drawing.Size(112, 24)
      Me.ChkKeepLinks.TabIndex = 2
      Me.ChkKeepLinks.Text = "Keep Links"
      '
      'Form1
      '
      Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
      Me.ClientSize = New System.Drawing.Size(288, 182)
      Me.Controls.AddRange(New System.Windows.Forms.Control() {Me.ChkKeepLinks, Me.Label1, Me.GroupBox2, Me.GroupBox1})
      Me.Name = "Form1"
      Me.Text = "SnagIT COM Example"
      Me.GroupBox1.ResumeLayout(False)
      Me.GroupBox2.ResumeLayout(False)
      Me.ResumeLayout(False)

   End Sub

#End Region

   Private Sub BtnWindow_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles BtnWindow.Click
      ' Choose an input and an output:
      SnagImg.Input = SNAGITLib.snagImageInput.siiWindow
      SnagImg.Output = SNAGITLib.snagImageOutput.sioFile

      ' Prompting for the file name is the default, but it cannot hurt to 
      ' set this explicitly
      SnagImg.OutputImageFile.FileNamingMethod = SNAGITLib.snagOuputFileNamingMethod.sofnmPrompt

      ' Show Preview Window?
      SnagImg.EnablePreviewWindow = ChkPreviewWindow.Checked

      ' Include cursor if set
      SnagImg.IncludeCursor = ChkIncludeCursor.Checked

      ''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
      ''' The following demonstrate new features in SnagIt 8.1

      ' Check keep links (hotspots)
      If ChkKeepLinks.Checked Then
         SnagImg.HotspotType = SNAGITLib.snagHotspotType.shtLinksOnly
         ' use a compatable file type by default
         SnagImg.OutputImageFile.FileType = SNAGITLib.snagImageFileType.siftSWF
      Else
         SnagImg.HotspotType = SNAGITLib.snagHotspotType.shtNone
         ' use PNG for a default file type
         SnagImg.OutputImageFile.FileType = SNAGITLib.snagImageFileType.siftPNG
      End If

      ' Place a caption outside the image:
      SnagImg.Filters.Annotation.EnableCaption = True
      SnagImg.Filters.Annotation.CaptionText = "Example Caption: Window!"
      SnagImg.Filters.Annotation.CaptionOptions.Placement = SNAGITLib.snagPlacement.spOutsideBottom

      ''' End of new 8.1 features
      ''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''

      ' Begin the capture process trapping any errors.
      Try
         SnagImg.Capture()
      Catch captureError As Exception
         If SnagImg.LastError = SNAGITLib.snagError.serrSnagItExpired Then
            MsgBox("Unable to capture: SnagIt evaluation has expired")
         End If
      End Try

   End Sub

   Private Sub BtnRegion_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles BtnRegion.Click

      ' Choose an input and an output:
      SnagImg.Input = SNAGITLib.snagImageInput.siiRegion
      SnagImg.Output = SNAGITLib.snagImageOutput.sioFile

      ' Prompting for the file name is the default, but it cannot hurt to set this explicitly
      SnagImg.OutputImageFile.FileNamingMethod = SNAGITLib.snagOuputFileNamingMethod.sofnmPrompt

      ' Show Preview Window?
      SnagImg.EnablePreviewWindow = ChkPreviewWindow.Checked

      ' Include cursor if set
      SnagImg.IncludeCursor = ChkIncludeCursor.Checked

      ''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
      ''' The following demonstrate new features in SnagIt 8.1

      ' Check keep links (hotspots)
      If ChkKeepLinks.Checked Then
         SnagImg.HotspotType = SNAGITLib.snagHotspotType.shtLinksOnly
         ' use a compatable file type by default
         SnagImg.OutputImageFile.FileType = SNAGITLib.snagImageFileType.siftSWF
      Else
         SnagImg.HotspotType = SNAGITLib.snagHotspotType.shtNone
         ' use PNG for a default file type
         SnagImg.OutputImageFile.FileType = SNAGITLib.snagImageFileType.siftPNG
      End If

      ' Place a caption outside the image:
      SnagImg.Filters.Annotation.EnableCaption = True
      SnagImg.Filters.Annotation.CaptionText = "Example Caption: Region!"
      SnagImg.Filters.Annotation.CaptionOptions.Placement = SNAGITLib.snagPlacement.spOutsideBottom

      ''' End of new 8.1 features
      ''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''

      ' Begin the capture process trapping any errors.
      Try
         SnagImg.Capture()
      Catch captureError As Exception
         If SnagImg.LastError = SNAGITLib.snagError.serrSnagItExpired Then
            MsgBox("Unable to capture: SnagIt evaluation has expired")
         End If
      End Try

   End Sub

End Class

File: Archives/CSharp/Simple/Form1.cs
//---------------------------------------------------------------------------
// Filename: Form1.cs - Contains the code for the form for the SnagIt COM
//           server C# simple sample.
//  
// Support e-mail: support@techsmith.zendesk.com  
//  
// Copyright © 2003-2014 TechSmith Corporation. All rights reserved.  
//---------------------------------------------------------------------------

using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;

namespace Simple
{
	/// <summary>
	/// Summary description for Form1.
	/// </summary>
	public class Form1 : System.Windows.Forms.Form
	{
      // Declare the SnagIt ImageCapture object.  NOTE: You must add a 
      // reference to the project in order for this line to compile.  
      private SNAGITLib.ImageCapture SnagImg;

      private System.Windows.Forms.Label Label1;
      private System.Windows.Forms.GroupBox GroupBox2;
      private System.Windows.Forms.Button BtnRegion;
      private System.Windows.Forms.Button BtnWindow;
      private System.Windows.Forms.GroupBox GroupBox1;
      private System.Windows.Forms.CheckBox ChkPreviewWindow;
      private System.Windows.Forms.CheckBox ChkIncludeCursor;
		/// <summary>
		/// Required designer variable.
		/// </summary>
		private System.ComponentModel.Container components = null;

		public Form1()
		{
			//
			// Required for Windows Form Designer support
			//
			InitializeComponent();

         // Create a new SnagIt ImageCapture object.
         SnagImg = new SNAGITLib.ImageCaptureClass();
		}

		/// <summary>
		/// Clean up any resources being used.
		/// </summary>
		protected override void Dispose( bool disposing )
		{
			if( disposing )
			{
				if (components != null) 
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
		}

		#region Windows Form Designer generated code
		/// <summary>
		/// Required method for Designer support - do not modify
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
         this.Label1 = new System.Windows.Forms.Label();
         this.GroupBox2 = new System.Windows.Forms.GroupBox();
         this.BtnRegion = new System.Windows.Forms.Button();
         this.BtnWindow = new System.Windows.Forms.Button();
         this.GroupBox1 = new System.Windows.Forms.GroupBox();
         this.ChkPreviewWindow = new System.Windows.Forms.CheckBox();
         this.ChkIncludeCursor = new System.Windows.Forms.CheckBox();
         this.GroupBox2.SuspendLayout();
         this.GroupBox1.SuspendLayout();
         this.SuspendLayout();
         // 
         // Label1
         // 
         this.Label1.Location = new System.Drawing.Point(16, 104);
         this.Label1.Name = "Label1";
         this.Label1.Size = new System.Drawing.Size(264, 40);
         this.Label1.TabIndex = 8;
         this.Label1.Text = "For simplicity, we will always output to a file, the name of which will be suppli" +
            "ed by the user when prompted.";
         // 
         // GroupBox2
         // 
         this.GroupBox2.Controls.AddRange(new System.Windows.Forms.Control[] {
                                                                                this.BtnRegion,
                                                                                this.BtnWindow});
         this.GroupBox2.Location = new System.Drawing.Point(8, 16);
         this.GroupBox2.Name = "GroupBox2";
         this.GroupBox2.Size = new System.Drawing.Size(128, 80);
         this.GroupBox2.TabIndex = 7;
         this.GroupBox2.TabStop = false;
         this.GroupBox2.Text = "Capture Type";
         // 
         // BtnRegion
         // 
         this.BtnRegion.Location = new System.Drawing.Point(16, 48);
         this.BtnRegion.Name = "BtnRegion";
         this.BtnRegion.Size = new System.Drawing.Size(96, 23);
         this.BtnRegion.TabIndex = 1;
         this.BtnRegion.Text = "Region Capture";
         this.BtnRegion.Click += new System.EventHandler(this.BtnRegion_Click);
         // 
         // BtnWindow
         // 
         this.BtnWindow.Location = new System.Drawing.Point(16, 16);
         this.BtnWindow.Name = "BtnWindow";
         this.BtnWindow.Size = new System.Drawing.Size(96, 24);
         this.BtnWindow.TabIndex = 0;
         this.BtnWindow.Text = "Window Capture";
         this.BtnWindow.Click += new System.EventHandler(this.BtnWindow_Click);
         // 
         // GroupBox1
         // 
         this.GroupBox1.Controls.AddRange(new System.Windows.Forms.Control[] {
                                                                                this.ChkPreviewWindow,
                                                                                this.ChkIncludeCursor});
         this.GroupBox1.Location = new System.Drawing.Point(144, 16);
         this.GroupBox1.Name = "GroupBox1";
         this.GroupBox1.Size = new System.Drawing.Size(136, 80);
         this.GroupBox1.TabIndex = 6;
         this.GroupBox1.TabStop = false;
         this.GroupBox1.Text = "Options";
         // 
         // ChkPreviewWindow
         // 
         this.ChkPreviewWindow.Location = new System.Drawing.Point(16, 48);
         this.ChkPreviewWindow.Name = "ChkPreviewWindow";
         this.ChkPreviewWindow.Size = new System.Drawing.Size(112, 24);
         this.ChkPreviewWindow.TabIndex = 2;
         this.ChkPreviewWindow.Text = "Preview Window";
         // 
         // ChkIncludeCursor
         // 
         this.ChkIncludeCursor.Location = new System.Drawing.Point(16, 24);
         this.ChkIncludeCursor.Name = "ChkIncludeCursor";
         this.ChkIncludeCursor.TabIndex = 1;
         this.ChkIncludeCursor.Text = "Include Cursor";
         // 
         // Form1
         // 
         this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
         this.ClientSize = new System.Drawing.Size(292, 158);
         this.Controls.AddRange(new System.Windows.Forms.Control[] {
                                                                      this.Label1,
                                                                      this.GroupBox2,
                                                                      this.GroupBox1});
         this.Name = "Form1";
         this.Text = "SnagIt COM Example";
         this.GroupBox2.ResumeLayout(false);
         this.GroupBox1.ResumeLayout(false);
         this.ResumeLayout(false);

      }
		#endregion

		/// <summary>
		/// The main entry point for the application.
		/// </summary>
		[STAThread]
		static void Main() 
		{
			Application.Run(new Form1());
		}

      private void BtnRegion_Click(object sender, System.EventArgs e)
      {
         // Choose an input and an output:
         SnagImg.Input = SNAGITLib.snagImageInput.siiRegion;
         SnagImg.Output = SNAGITLib.snagImageOutput.sioFile;

         // Prompting for the file name is the default, but it cannot hurt to set this explicitly
         SnagImg.OutputImageFile.FileNamingMethod = SNAGITLib.snagOuputFileNamingMethod.sofnmPrompt;

         // Show Preview Window?
         SnagImg.EnablePreviewWindow = ChkPreviewWindow.Checked;

         // Include cursor if set
         SnagImg.IncludeCursor = ChkIncludeCursor.Checked;

         // Try to initiate the capture.. catch any errors and display an 
         // appropriate error mesasge.  Here, SnagIt expiration error is shown
         // as an example.
         try
         {
            SnagImg.Capture();
         }
         catch ( Exception exception )
         {
            if ( SnagImg.LastError == SNAGITLib.snagError.serrSnagItExpired )
            {
               MessageBox.Show( "Unable to capture: SnagIt evaluation has expired" );
            }
         }
      }

      private void BtnWindow_Click(object sender, System.EventArgs e)
      {
         // Choose an input and an output:
         SnagImg.Input = SNAGITLib.snagImageInput.siiWindow;
         SnagImg.Output = SNAGITLib.snagImageOutput.sioFile;

         // Prompting for the file name is the default, but it cannot hurt to 
         // set this explicitly
         SnagImg.OutputImageFile.FileNamingMethod = SNAGITLib.snagOuputFileNamingMethod.sofnmPrompt;

         // Show Preview Window?
         SnagImg.EnablePreviewWindow = ChkPreviewWindow.Checked;

         // Include cursor if set
         SnagImg.IncludeCursor = ChkIncludeCursor.Checked;
      
         // Try to initiate the capture.. catch any errors and display an 
         // appropriate error mesasge.  Here, SnagIt expiration error is shown
         // as an example.
         try
         {
            SnagImg.Capture();
         }
         catch ( Exception exception )
         {
            if ( SnagImg.LastError == SNAGITLib.snagError.serrSnagItExpired )
            {
               MessageBox.Show( "Unable to capture: SnagIt evaluation has expired" );
            }
         }
      }
	}
}
